File: hw_06/Python/hw_06/clothes/Shirt.py
from clothes.Color import Color
from clothes.Size import Size

class Shirt:
    def __init__(self, color:Color=Color.Black, size:Size=Size.L, price:float=199.00):
        self.__color = color
        self.__size = size
        self.__price = price

    def get_color(self):
        return self.__color
    
    def set_color(self, color:Color):
        self.__color = color

    def get_size(self):
        return self.__size
    
    def set_size(self, size:Size):
        self.__size = size

    def get_price(self):
        return self.__price
    
    def set_price(self, price:float):
        self.__price = price

    def info(self):
        return f"Shirt: Color {self.__color.name}, Size {self.__size.name}, Price {self.__price}" 

File: hw_06/CPP/cpp_oppwork06/clothes/shirt.cpp
#include <iostream>
#include <map>
#include "color.cpp"
#include "size.cpp"

using namespace std;

class Shirt
{
private:
    Color color;
    Size size;
    double price;

public:
    Shirt(Color color, Size size, double price) : color(color), size(size), price(price) {}

    Shirt() : Shirt(Color::Black, Size::L, 199.00) {}

    Color getColor()
    {
        return color;
    }

    void setColor(Color color)
    {
        this->color = color;
    }

    string toString(Color c)
    {
        static map<Color, string> colorMap = {
            {Color::Black, "Black"},
            {Color::White, "White"},
            {Color::Red, "Red"},
            {Color::Green, "Green"},
            {Color::Blue, "Blue"}};

        return colorMap[c];
    }

    string toString(Size s)
    {
        static map<Size, string> sizeMap = {
            {Size::S, "S"},
            {Size::M, "M"},
            {Size::L, "L"},
            {Size::XL, "XL"}};

        return sizeMap[s];
    }

    void info()
    {
        cout << "Shirt: [ Color " <<  toString(color) << ", Size " << toString(size) << ", Price " << price << " ]" << endl;
    }
};